### ISSUE TITLE
ETTh2-96 论文结果无法通过官方脚本稳定复现，请说明 checkpoint 选择与评估口径

### ISSUE THREAD

**Issue by kaka-lz** [saber1360/SEED#3]
```markdown
我使用本仓库官方脚本 `scripts/ETTh2.sh` 复现论文 Table A3 中 SEED 在 ETTh2 96->96 上的结果，但结果与论文报告值存在明显差异。

论文 Table A3 报告：

```text
ETTh2 96->96
MSE: 0.282
MAE: 0.336
```

我使用官方脚本中的 ETTh2 96->96 配置：

```bash
python -u run.py \
  --task_name long_term_forecast \
  --is_training 1 \
  --root_path ./data \
  --data_path ETTh2.csv \
  --model_id ETTh2_96_96 \
  --model SEED \
  --data ETTh2 \
  --features M \
  --seq_len 96 \
  --label_len 48 \
  --pred_len 96 \
  --e_layers 1 \
  --d_layers 1 \
  --factor 3 \
  --enc_in 7 \
  --dec_in 7 \
  --c_out 7 \
  --patch_len 16 \
  --dropout 0.8 \
  --top_p 0.0 \
  --des Exp \
  --learning_rate 0.0001 \
  --batch_size 32 \
  --train_epochs 15 \
  --d_model 128 \
  --d_ff 256 \
  --itr 1 \
  --use_revin True \
  --use_dropout 0.05
```

环境：

```text
Python 3.10.16
torch 2.3.1+cu121
numpy 1.26.4
pandas 2.2.3
scikit-learn 1.5.2
```

训练日志如下。可以看到，训练过程中打印的 `Test Loss` 曾接近论文结果，但最终 `test()` 输出的 full-sample metric 明显更差。

```text
Epoch: 1, Steps: 265 | Train Loss: 0.4747856 Vali Loss: 0.2208139 Test Loss: 0.2932566
Validation loss decreased (inf --> 0.220814).  Saving model ...

Epoch: 2, Steps: 265 | Train Loss: 0.4354076 Vali Loss: 0.2193601 Test Loss: 0.2895655
Validation loss decreased (0.220814 --> 0.219360).  Saving model ...

Epoch: 3, Steps: 265 | Train Loss: 0.4104231 Vali Loss: 0.2215775 Test Loss: 0.2920573
Validation loss decreased (0.219360 --> 0.221577).  Saving model ...

Epoch: 4, Steps: 265 | Train Loss: 0.3934230 Vali Loss: 0.2222966 Test Loss: 0.2878638
Validation loss decreased (0.221577 --> 0.222297).  Saving model ...

Epoch: 5, Steps: 265 | Train Loss: 0.3865503 Vali Loss: 0.2149187 Test Loss: 0.2907653
Validation loss decreased (0.222297 --> 0.214919).  Saving model ...

Epoch: 6, Steps: 265 | Train Loss: 0.3809946 Vali Loss: 0.2201452 Test Loss: 0.2819860
Validation loss decreased (0.214919 --> 0.220145).  Saving model ...

Epoch: 7, Steps: 265 | Train Loss: 0.3774195 Vali Loss: 0.2126694 Test Loss: 0.2899894
Validation loss decreased (0.220145 --> 0.212669).  Saving model ...

Epoch: 8, Steps: 265 | Train Loss: 0.3736686 Vali Loss: 0.2166144 Test Loss: 0.2802847
Validation loss decreased (0.212669 --> 0.216614).  Saving model ...

Epoch: 9, Steps: 265 | Train Loss: 0.3711626 Vali Loss: 0.2135570 Test Loss: 0.2831604
Validation loss decreased (0.216614 --> 0.213557).  Saving model ...

Epoch: 10, Steps: 265 | Train Loss: 0.3701340 Vali Loss: 0.2136465 Test Loss: 0.2888693
Validation loss decreased (0.213557 --> 0.213647).  Saving model ...

Epoch: 11, Steps: 265 | Train Loss: 0.3673378 Vali Loss: 0.2139523 Test Loss: 0.2858223
Validation loss decreased (0.213647 --> 0.213952).  Saving model ...

Epoch: 12, Steps: 265 | Train Loss: 0.3821911 Vali Loss: 0.2138016 Test Loss: 0.2847926
Validation loss decreased (0.213952 --> 0.213802).  Saving model ...

Epoch: 13, Steps: 265 | Train Loss: 0.3662261 Vali Loss: 0.2132146 Test Loss: 0.2843627
Validation loss decreased (0.213802 --> 0.213215).  Saving model ...

Epoch: 14, Steps: 265 | Train Loss: 0.3651605 Vali Loss: 0.2150515 Test Loss: 0.2851485
Validation loss decreased (0.213215 --> 0.215051).  Saving model ...

Epoch: 15, Steps: 265 | Train Loss: 0.3633172 Vali Loss: 0.2135558 Test Loss: 0.2852180
Validation loss decreased (0.215051 --> 0.213556).  Saving model ...

Final test:
mse: 0.2870101
mae: 0.3406565
```

这里有几个问题需要澄清：

1. 论文 Table A3 的 `0.282 / 0.336` 是怎么得到的？  
   这个结果无法通过官方脚本最终 `test()` 输出稳定复现。

2. 论文是否使用了训练过程中某个 epoch 的 `Test Loss` 作为最终结果？  
   因为日志中的 `Epoch 6/8 Test Loss` 接近论文结果，但最终 `test()` 输出明显不同。

3. `exp_long_term_forecasting.py` 中训练过程的 `vali()` 是按 batch 平均 loss，而最终 `test()` 是拼接所有样本后计算 metric。请问论文表格使用的是哪一种评估口径？

4. 官方脚本中 `patience` 默认是 `-1`，这会导致每个 epoch 都覆盖保存 checkpoint，并不是 validation-best checkpoint。论文表格使用的是 final epoch、validation-best epoch，还是 test-best epoch？

5. 训练日志中出现了如下输出：

```text
Validation loss decreased (0.212669 --> 0.216614). Saving model ...
```

但这里 validation loss 实际变大了。这是因为 `patience=-1` 时每轮强制保存 checkpoint。请问论文实验是否也是这个设置？

6. 请提供能够复现 Table A3 中 ETTh2 96->96 `0.282 / 0.336` 的完整脚本、checkpoint 选择规则，或对应 checkpoint。

目前按照仓库官方脚本和最终 full-sample metric，我得到的是：

```text
MSE: 0.2870101
MAE: 0.3406565
```

这与论文报告结果存在差异。希望作者能明确说明论文结果的 checkpoint 选择方式和评估口径。
**（该文展示在各个数据集都取得了不错的效果,思路也比较清晰。
请作者做出详细的回复或缺失补充，供后续研究者学习。若有设计漏洞也请及时指出，以免误导研究者🌹🌹🌹）**

**Comment by saber1360** [comment]
[etth2_mut_auto.log](https://github.com/user-attachments/files/27556362/etth2_mut_auto.log)

这是当初实验的log，代码确实已经不知道是哪个版本了，这个仓库是后期整理过后的，可能选错了备份或者遗漏了一些东西，cpkt选取的最终的结果啊，谁会用测试最好的ckpt的结果啊，那也算有一些数据泄露了吧，实验部分使用的时序预测祖传的exp_long_term_forecasting.py，稍微改写了一丢丢，patience=-1为弃用早停法，了解下早停法吧earlystop

<img width="1197" height="201" alt="Image" src="https://github.com/user-attachments/assets/128d5871-22ff-494b-ba29-f9d51479cd8e" />
